bemani/utils/read.py

```diff
@@ -1480,6 +1509,41 @@ def lookup(self, server: str, token: str) -> Tuple[List[Dict[str, Any]], List[Di
 
         return reassembled_songs, emblems
 
+    def __revivals(self, songid: int, chart: int) -> Optional[int]:
+        old_id = self.get_music_id_for_song(songid, chart)
+        if old_id is not None:
+            return old_id
+
+        # In qubell and clan omnimix, PPAP and Bonjour the world are placed
+        # at this arbitrary songid since they weren't assigned one originally
+        # In jubeat festo, these songs were given proper songids so we need to account for this
+        legacy_to_modern_map = {
+            71000001: 70000124,  # PPAP
+            71000002: 70000154,  # Bonjour the world
+            50000020: 80000037,  # 千本桜 was removed and then revived in clan
+            60000063: 70000100,  # Khamen break sdvx had the first id for prop(never released officially)
+        }
+
+        legacy_songid = legacy_to_modern_map.get(songid)
+        if legacy_songid is not None:
+            old_id = self.get_music_id_for_song(legacy_songid, chart)
+            if old_id is not None:
+                return old_id
+        modern_to_legacy_map = {
```

Comment:
I see no good reason not to create this by walking the `legacy_to_modern_map` in a comprehension and just swapping the key/value pair, like `{v, k for k, v in legacy_to_modern_map.items()}`.